src/components/extra-questions/index.js

```diff
@@ -0,0 +1,326 @@
+                  value={props.input.value}
+                  options={questionValues}
+                  onChange={props.input.onChange}
+                />
+              )}
+            </Field>
+          </div>
+          {q.sub_question_rules?.length > 0 &&
+            q.sub_question_rules.map((r) => {
+              return (
+                <Condition when={q.name} rule={r}>
+                  {renderQuestion(r.sub_question)}
+                </Condition>
+              );
+            })}
+        </>
+      );
+    }
+    if (q.type === "CheckBoxList") {
+      questionValues = questionValues.map(val => ({ ...val, value: val.id }));
+      return (
+        <>
+          <div key={q.id} ref={el => questionRef.current[q.id] = el}>
+            <RawHTML>{q.mandatory ? q.label?.endsWith('</p>') ? q.label.replace(/<\/p>$/g, " <b>*</b></p>") : `${q.label} <b>*</b>` : q.label}</RawHTML>
+            <Field name={q.name}>
+              {props => (
+                <CheckboxList
+                  id={q.id}
+                  value={props.input.value}
+                  options={questionValues}
+                  onChange={props.input.onChange}
+                />
+              )}
+            </Field>
+          </div>
+          {q.sub_question_rules?.length > 0 &&
+            q.sub_question_rules.map((r) => {
+              return (
+                <Condition when={q.name} rule={r}>
+                  {renderQuestion(r.sub_question)}
+                </Condition>
+              );
+            })}
+        </>
+      );
+    }
+    return null;
+  };
+
+  const isVisible = (q) => {
+    return !!questionRef.current[q.id];
+  };
+
+  const validateQuestion = (q, values, errors) => {
+    if (q.mandatory && isVisible(q)) {
+      if (!values.hasOwnProperty(q.name) || values[q.name] === "") {
+        errors[q.name] = "Required";
+      }
+    }
+    // validate sub rules
+    q.sub_question_rules?.forEach((r) => {
+      validateQuestion(r.sub_question, values, errors);
+    });
+  };
+
+  const onSubmit = (values) => {
+    onAnswerChanges(values)
+  };
+
+  return (
+    <div className={className}>
+      <Form
+        onSubmit={onSubmit}
+        initialValues={answers}
+        validate={(values) => {
+          const errors = {};
+          extraQuestions.map((q) => {
+            validateQuestion(q, values, errors);
+          });
+          return errors;
+        }}
+      >
+        {({ handleSubmit, form, submitting, pristine, values }) => {
+          submit = handleSubmit;
+          return (
+            <form onSubmit={handleSubmit}>
+              {extraQuestions.map((q) => renderQuestion(q, questionRef[q.id]))}
+              {debug && <pre>{JSON.stringify(values, 0, 2)}</pre>}
+            </form>
+          );
+        }}
+      </Form>
+      <button
+        type="submit"
+        onClick={(event) => {
+          submit(event);
+        }}
+        style={{ marginTop: 10 }}
+      >
+        {buttonText}
```

Comment:
@tomrndom remember that we need to handle the case of the external submit button too